NWM Client New Test Failure: AttributeError: 'EntryPoints' object has no attribute 'get'

The test runners for #210 found a [failing `nwm_client_new` test](https://github.com/NOAA-OWP/hydrotools/actions/runs/3500004127/jobs/5862205193#step:5:39). The `AttributeError` is raised in `xarray` source when getting a `xarray` backend. See the below collapsed test failure trace for more detail.

```shell
FAILED python/nwm_client_new/tests/test_NWMFileProcessor.py::test_get_dataset - AttributeError: 'EntryPoints' object has no attribute 'get'
```
<details>
<summary>
full trace
</summary>

[source](https://github.com/NOAA-OWP/hydrotools/actions/runs/3500004127/jobs/5862205193#step:5:39)

```shell
=================================== FAILURES ===================================
_______________________________ test_get_dataset _______________________________

Unclosed client session
client_session: <hydrotools._restclient.async_client.ClientSession object at 0x7f90b51bf650>
    def test_get_dataset():
>       ds = NWMFileProcessor.get_dataset(input_directory)

python/nwm_client_new/tests/test_NWMFileProcessor.py:8: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/hydrotools/nwm_client_new/NWMFileProcessor.py:59: in get_dataset
    ds = xr.open_mfdataset(file_list, engine="netcdf4")
/opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/xarray/backends/api.py:908: in open_mfdataset
    datasets = [open_(p, **open_kwargs) for p in paths]
/opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/xarray/backends/api.py:908: in <listcomp>
    datasets = [open_(p, **open_kwargs) for p in paths]
/opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/xarray/backends/api.py:[48](https://github.com/NOAA-OWP/hydrotools/actions/runs/3500004127/jobs/5862205193#step:5:49)1: in open_dataset
    backend = plugins.get_backend(engine)
/opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/xarray/backends/plugins.py:161: in get_backend
    engines = list_engines()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    @functools.lru_cache(maxsize=1)
    def list_engines():
>       entrypoints = entry_points().get("xarray.backends", ())
E       AttributeError: 'EntryPoints' object has no attribute 'get'

/opt/hostedtoolcache/Python/3.7.15/x[64](https://github.com/NOAA-OWP/hydrotools/actions/runs/3500004127/jobs/5862205193#step:5:65)/lib/python3.7/site-packages/xarray/backends/plugins.py:105: AttributeError
=============================== warnings summary ===============================
python/nwm_client_new/tests/test_NWMClient.py::test_QueryError
  /opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/hydrotools/nwm_client_new/NWMFileClient.py:197: RuntimeWarning: No data found for configuration 'analysis_assim' and reference time '20T00Z'
    warnings.warn(message, RuntimeWarning)

python/nwm_client_new/tests/test_NWMClient.py::test_QueryError
  /opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/hydrotools/nwm_client_new/NWMFileClient.py:197: RuntimeWarning: No data found for configuration 'analysis_assim' and reference time '30T01Z'
    warnings.warn(message, RuntimeWarning)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED python/nwm_client_new/tests/test_NWMFileProcessor.py::test_get_dataset - AttributeError: 'EntryPoints' object has no attribute 'get'
```

</details>

A quick look on `xarray`'s issue tracker led me to [this](https://github.com/pydata/xarray/issues/7149) -- a verbatim reproduction of this issue. Conversation there pointed to the issue's source, the newly released version (`5.0.0`) of `importlib-metadata` and `python 3.7` (no longer actively supported by `xarray`).

--- a/python/nwm_client_new/setup.cfg
+++ b/python/nwm_client_new/setup.cfg
@@ -41,6 +41,8 @@ install_requires =
     aiofiles
     netcdf4
     tables
+    # remove when 3.7 support is dropped
+    importlib-metadata<=4.13.0
 python_requires = >=3.7
 include_package_data = True
 

